### DISCUSSION TITLE
Photoprism gives space error and refused to start

### DISCUSSION THREAD

**Discussion by tmzhuang** [photoprism/photoprism#4202] [Q&A]
In my system journal (journalctl -b -u photoprism), I saw some lines like:
```
Apr 19 14:50:22 pluto photoprism[377]: time="2024-04-19T14:50:22-04:00" level=error msg="moments: Error 1021: Disk full (/tmp/#sql-temptable-1c9-201-61.MAI); waiting for someone to free some space... (errno: 28 \"No space left on device\")"
Apr 19 14:50:25 pluto photoprism[377]: time="2024-04-19T14:50:25-04:00" level=error msg="moments: Error 1021: Disk full (/tmp/#sql-temptable-1c9-203-80.MAI); waiting for someone to free some space... (errno: 28 \"No space left on device\") (update folder dates)"
Apr 19 14:50:25 pluto photoprism[377]: time="2024-04-19T14:50:25-04:00" level=error msg="moments: Error 1021: Disk full (/tmp/#sql-temptable-1c9-201-63.MAI); waiting for someone to free some space... (errno: 28 \"No space left on device\") (update album dates)"
```

I took a look in the logs because photos stopped being displayed in the browser in my photoprism instance. I took at look on the server and there didn't seem to be any space or RAM space issues. Restarting photoprism gives a return code of 1 with no other output to stdout and without the photoprism server starting.

Can anyone help me solve this?

**Comment by lastzero** [comment]
It seems like the `/tmp` path in the Docker container is full. Since MariaDB does not use this path from all I know, could it be SQLite? 

I recommend checking all your disks (not just the one where your pictures are stored) and if that doesn't help, try to configure a different temp path or use MariaDB:
- https://docs.photoprism.app/getting-started/config-options/#storage
- https://docs.photoprism.app/getting-started/troubleshooting/docker/#disk-space
- https://docs.photoprism.app/getting-started/troubleshooting/#fatal-server-errors

**Reply by tmzhuang** [reply]
Ah you are right. I'm actually not on docker but my /tmp is a tmpfs partition and it's full. Is 4G too small for my /tmp directory? There's only 16G total on the system. I do notice that there are a large number of photoprism directories inside of temp that haven't been cleaned up despite photoprism not running.
